weed/filer/etcd/etcd_store.go

```diff
@@ -164,8 +164,10 @@ func (store *EtcdStore) ListDirectoryEntries(ctx context.Context, dirPath weed_u
 		return lastFileName, fmt.Errorf("list %s : %v", dirPath, err)
 	}
 
+	etcdKeyWithDirectoryPrefix := append([]byte(store.etcdKeyPrefix), directoryPrefix...)
```

Comment:
how about adding a `if/else` check to avoid the append operation if `store.etcdKeyPrefix` is empty.